public/app/features/explore/NodeGraph/NodeGraphContainer.tsx

```diff
@@ -73,35 +75,42 @@ export function UnconnectedNodeGraphContainer(props: Props) {
       setTop(top);
     }
   }, [containerRef]);
-  const height = windowHeight - top - 32;
+  const height = Math.max(600, windowHeight - top);
 
   const countWarning =
     withTraceView && nodes[0]?.length > 1000 ? (
       <span className={styles.warningText}> ({nodes[0].length} nodes, can be slow to load)</span>
-    ) : null;
+    ) : (
+      ''
+    );
+
+  const [measureRef, { width: measureWidth }] = useMeasure<HTMLDivElement>();
+
+  // Merging refs since we are using window size for height and div size for width
+  const refs = useCallback(
+    (node: HTMLDivElement | null) => {
+      if (node !== null) {
+        (containerRef as React.MutableRefObject<HTMLDivElement | null>).current = node;
+        measureRef(node);
+      }
+    },
+    [measureRef]
+  );
 
   return (
-    <Collapse
-      label={<span>Node graph{countWarning} </span>}
-      collapsible={withTraceView}
-      // We allow collapsing this only when it is shown together with trace view.
-      isOpen={withTraceView ? open : true}
-      onToggle={withTraceView ? () => toggled() : undefined}
-    >
-      <div
-        ref={containerRef}
-        style={
-          withTraceView
-            ? { height: 500 }
-            : {
-                minHeight: 600,
-                height,
-              }
-        }
+    <div ref={refs}>
+      <PanelChrome
+        title={`Node graph${countWarning}`}
+        width={measureWidth}
```

Comment:
We keep `containerWidth` in explore state to handle the pane resizing, did you look into using thatinstead of a separate hook?

Reply:
After looking into this, it turns out that `containerWidth` takes the whole Explore container width, while the 'measureWidth` always takes the size of the NodeGraphContainer. We could use `containerWidth`, but would have to adjust it with something like `containerWidth - 32` because measureWidth or add padding. This is what happens to NodeGraph container width when passing containerWidth:

![Screenshot 2023-06-06 at 12 04 23 PM](https://github.com/grafana/grafana/assets/58232930/ba96a93c-21b1-40bc-9ba8-1481281d9779)